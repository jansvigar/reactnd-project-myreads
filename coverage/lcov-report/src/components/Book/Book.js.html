<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Book/Book.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/Book</a> Book.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.55% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>7/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.55% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/11</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import BookShelfChanger from './BookShelfChanger';
import PropTypes from 'prop-types';
import Rating from '../Rating/Rating'
import * as BooksAPI from '../../BooksAPI';
import './Book.css';
&nbsp;
class Book extends Component {
  state = {
      shelf: ''
  };
&nbsp;
  componentWillMount() {
    <span class="missing-if-branch" title="else path not taken" >E</span>if(this.props.book.shelf) {
      this.setState({ shelf: this.props.book.shelf });
    } else {
<span class="cstat-no" title="statement not covered" >      BooksAPI</span>
        .get(this.props.book.id)
        .then(<span class="fstat-no" title="function not covered" >bo</span>ok =&gt;
<span class="cstat-no" title="statement not covered" >          this.setState({ shelf: book.shelf })</span>
        );
    }
  };
&nbsp;
  handleOnChange = <span class="fstat-no" title="function not covered" >(s</span>electedValue) =&gt; {;
<span class="cstat-no" title="statement not covered" >    BooksAPI.update(this.props.book, selectedValue).then(<span class="fstat-no" title="function not covered" >(d</span>ata) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ shelf: selectedValue });</span>
<span class="cstat-no" title="statement not covered" >      this.props.updateShelvedBooksList</span>
        &amp;&amp; this.props.updateShelvedBooksList();
    });
  }
&nbsp;
  render(){
    const bookCoverStyles = {
      width: 128,
      height: 193,
      backgroundImage: `url(${this.props.book.imageLinks
                              &amp;&amp; this.props.book.imageLinks.thumbnail})`
    };
&nbsp;
    return (
      &lt;div className="book"&gt;
        &lt;div className="book-top"&gt;
          &lt;div className="book-cover" style={ bookCoverStyles }&gt;&lt;/div&gt;
          &lt;BookShelfChanger
            selectedShelf={ this.state.shelf }
            handleOnChange={ this.handleOnChange }
          /&gt;
        &lt;/div&gt;
        &lt;div className="book-rating"&gt;
          &lt;Rating number={ this.props.book.averageRating } /&gt;
        &lt;/div&gt;
        &lt;div className="book-title"&gt;{ this.props.book.title || <span class="branch-1 cbranch-no" title="branch not covered" >'Unknown Title' </span>}&lt;/div&gt;
        &lt;div className="book-authors"&gt;
          { this.props.book.authors &amp;&amp; this.props.book.authors.length &gt; 0
              ? this.props.book.authors.reduce((acc, cur) =&gt; `${acc}, ${cur}`)
              : <span class="branch-1 cbranch-no" title="branch not covered" >'Unknown Author' </span>}
        &lt;/div&gt;
      &lt;/div&gt;
    );
  };
}
&nbsp;
Book.propTypes = {
  book: PropTypes.shape ({
    id: PropTypes.string.isRequired,
    imageLinks: PropTypes.object,
    title: PropTypes.string,
    authors: PropTypes.array,
    averageRating: PropTypes.number,
    shelf: PropTypes.string
  })
};
&nbsp;
export default Book;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 11 2017 00:33:58 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
