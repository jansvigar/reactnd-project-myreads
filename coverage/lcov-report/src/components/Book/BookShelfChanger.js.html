<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Book/BookShelfChanger.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/Book</a> BookShelfChanger.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import * as Constants from '../../constants';
import { camelCase } from '../../utils';
&nbsp;
const BookShelfChanger = props =&gt; (
  &lt;div className="book-shelf-changer"&gt;
    &lt;select
      value={ props.selectedShelf || <span class="branch-1 cbranch-no" title="branch not covered" >'none' </span>}
      onChange={ <span class="fstat-no" title="function not covered" >e </span>=&gt; <span class="cstat-no" title="statement not covered" >props.handleOnChange(e.target.value) </span>}
    &gt;
      &lt;option value="" disabled&gt;Move to...&lt;/option&gt;
      {
        Constants.SHELVES.map(shelf =&gt; (
          &lt;option key={ shelf } value={ camelCase(shelf) }&gt;
            { shelf }
          &lt;/option&gt;
        ))
      }
      &lt;option value="none"&gt;None&lt;/option&gt;
    &lt;/select&gt;
  &lt;/div&gt;
);
&nbsp;
BookShelfChanger.propTypes = {
  selectedShelf: PropTypes.string,
  handleOnChange: PropTypes.func.isRequired
};
&nbsp;
export default BookShelfChanger;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 11 2017 00:33:58 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
