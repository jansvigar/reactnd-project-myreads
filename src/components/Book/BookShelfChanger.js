import React from 'react';
import PropTypes from 'prop-types';
import * as Constants from '../../constants';
import { camelCase } from '../../utils';

const BookShelfChanger = props => (
  <div className="book-shelf-changer">
    <select
      value={ props.selectedShelf || 'none' }
      onChange={ e => props.handleOnChange(e.target.value) }
    >
      <option value="" disabled>Move to...</option>
      {
        Constants.SHELVES.map(shelf => (
          <option key={ shelf } value={ camelCase(shelf) }>
            { shelf }
          </option>
        ))
      }
      <option value="none">None</option>
    </select>
  </div>
);

BookShelfChanger.propTypes = {
  selectedShelf: PropTypes.string,
  handleOnChange: PropTypes.func.isRequired
};

export default BookShelfChanger;
