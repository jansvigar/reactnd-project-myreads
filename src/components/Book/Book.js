import React from 'react';
import BookShelfChanger from './BookShelfChanger';
import PropTypes from 'prop-types';
import './Book.css';

const Book = props => {
  const { book :
            {
              imageLinks : { thumbnail } = {},
              title = 'Unknown Title',
              authors = ['Unknown Author']
            } = {}
        } = props;

  const bookCoverStyles = {
    width: 128,
    height: 193,
    backgroundImage: `url(${thumbnail})`
  };

  return (
    <div className="book">
      <div className="book-top">
        <div className="book-cover" style={ bookCoverStyles }></div>
        <BookShelfChanger />
      </div>
      <div className="book-title">{ title }</div>
      <div className="book-authors">
        { authors.reduce((acc, cur) => `${acc}, ${cur}`) }
      </div>
    </div>
  );
};

Book.propTypes = {
  book: PropTypes.shape ({
    imageLinks: PropTypes.object,
    title: PropTypes.string,
    authors: PropTypes.array
  })
};

export default Book;
