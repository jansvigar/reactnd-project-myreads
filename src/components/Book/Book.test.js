import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Book from './Book';
import BookShelfChanger from './BookShelfChanger'

const bookProps = {
  book: {
    id: 'id.123',
    imageLinks: {},
    title: 'Test Title',
    authors: ['author 1', 'author 2'],
    averageRating: 5,
    shelf: 'Test Shelf'
  }
};

const handleOnChange = jest.fn();

const bookShelfChangerProps = {
  selectedShelf: 'Test Shelf',
  handleOnChange: handleOnChange
}

describe('<Book />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Book {...bookProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('<BookShelfChanger />', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<BookShelfChanger {...bookShelfChangerProps} />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

});
