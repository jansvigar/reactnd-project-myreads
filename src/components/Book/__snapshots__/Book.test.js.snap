// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> <BookShelfChanger /> should render correctly 1`] = `
<div
  className="book-shelf-changer"
>
  <select
    onChange={[Function]}
    value="Test Shelf"
  >
    <option
      disabled={true}
      value=""
    >
      Move to...
    </option>
    <option
      value="currentlyReading"
    >
      Currently Reading
    </option>
    <option
      value="wantToRead"
    >
      Want to Read
    </option>
    <option
      value="read"
    >
      Read
    </option>
    <option
      value="none"
    >
      None
    </option>
  </select>
</div>
`;

exports[`<Book /> should render correctly 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(undefined)",
          "height": 193,
          "width": 128,
        }
      }
    />
    <BookShelfChanger
      handleOnChange={[Function]}
      selectedShelf="Test Shelf"
    />
  </div>
  <div
    className="book-rating"
  >
    <Rating
      number={5}
    />
  </div>
  <div
    className="book-title"
  >
    Test Title
  </div>
  <div
    className="book-authors"
  >
    author 1, author 2
  </div>
</div>
`;
